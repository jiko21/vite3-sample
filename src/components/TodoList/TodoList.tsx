import styled from '@emotion/styled';
import { Todo } from '../../todo';
import { Card } from '../Card';
import { Button } from '../Button';

const Ul = styled.ul`
  list-style: none;
  padding: 0;
`;

const Li = styled.li`
  margin: 4px 0;
`;

const CardContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
`;

const Span = styled.span`
  color: #3c3c3c;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

type Props = {
  todos: Todo[];
  onRemove: (id: number) => void;
  testId?: string;
};

export const TodoList: React.FC<Props> = ({ todos, onRemove, testId }) => {
  return (
    <Ul>
      {todos.map((todo) => (
        <Li key={todo.id}>
          <Card>
            <CardContainer>
              <Span>{todo.content}</Span>
              <Button
                onClick={() => onRemove(todo.id)}
                buttonStyle="danger"
                testId={`${testId}-remove-${todo.id}`}
              >
                remove
              </Button>
            </CardContainer>
          </Card>
        </Li>
      ))}
    </Ul>
  );
};
