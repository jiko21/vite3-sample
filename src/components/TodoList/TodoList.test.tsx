import { afterEach, describe, expect, it, vi } from 'vitest';
import { TodoList } from './TodoList';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';

const todos = [
  {
    id: 1,
    content: 'aaa',
  },
  {
    id: 2,
    content: 'bbb',
  },
];

describe('components/TodoList', () => {
  afterEach(cleanup);

  it('correctly render', () => {
    const onRemoveMock = vi.fn();
    const { container } = render(<TodoList testId="todoList" onRemove={onRemoveMock} todos={todos} />);
    expect(container).toMatchSnapshot();
  });

  it('correctly call remove', () => {
    const onRemoveMock = vi.fn();
    const { container } = render(<TodoList testId="todoList" onRemove={onRemoveMock} todos={todos} />);
    fireEvent.click(screen.getByTestId('todoList-remove-2-button'));
    expect(onRemoveMock).toBeCalledWith(2);
  });
});
