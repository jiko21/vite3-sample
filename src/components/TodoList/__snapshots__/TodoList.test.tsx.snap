// Vitest Snapshot v1

exports[`components/TodoList > correctly render 1`] = `
<div>
  <ul
    class="css-gflyic"
  >
    <li
      class="css-ux9ziv"
    >
      <div
        class="css-1b7m767-Card"
      >
        <div
          class="css-1vo4ybp"
        >
          <span
            class="css-10rfslh"
          >
            aaa
          </span>
          <button
            class="css-gu30kj-Button"
            data-testid="todoList-remove-1-button"
          >
            remove
          </button>
        </div>
      </div>
    </li>
    <li
      class="css-ux9ziv"
    >
      <div
        class="css-1b7m767-Card"
      >
        <div
          class="css-1vo4ybp"
        >
          <span
            class="css-10rfslh"
          >
            bbb
          </span>
          <button
            class="css-gu30kj-Button"
            data-testid="todoList-remove-2-button"
          >
            remove
          </button>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
